### PR TITLE
[RFC0028] Enable using protected CNBs

### PR DESCRIPTION
**Pull request by modulo11**
Thanks for contributing to cloud_controller_ng. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:

Building on https://github.com/cloudfoundry/cloud_controller_ng/pull/3778, this PR introduces the ability to use the `cnb` lifecycle with password protected Cloud Native Buildpacks.

* An explanation of the use cases your change solves

* Links to any other associated PRs

* [X] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [X] I have viewed, signed, and submitted the Contributor License Agreement

* [X] I have made this pull request to the `main` branch

* [X] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/spec/README.md#cf-acceptance-tests-cats)

### CODE REVIEW

File: app/models/runtime/cnb_lifecycle_data_model.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 require 'cloud_controller/diego/lifecycles/lifecycles'
-
+require 'presenters/helpers/censorship'
```
Comment (pbusko) [inline app/models/runtime/cnb_lifecycle_data_model.rb:2]:
```suggestion
require 'presenters/helpers/censorship'

```

File: spec/unit/lib/cloud_controller/diego/lifecycle_data_spec.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 require 'spec_helper'
```
Comment (c0d1ngm0nk3y) [inline spec/unit/lib/cloud_controller/diego/lifecycle_data_spec.rb:1]: Shouldn't that file be located at `spec/unit/lib/cloud_controller/diego/buildpack/lifecycle_data_spec.rb`

**Review by johha — CHANGES_REQUESTED** [review]
Looks good overall, just found one minor thing.

I'm a bit unsure if we should have two `lifecycle_data.rb` classes but also don't have a better idea at the moment

File: app/models/runtime/cnb_lifecycle_data_model.rb [hunk 3]
```diff
@@ -1,8 +1,10 @@
 require 'cloud_controller/diego/lifecycles/lifecycles'
+require 'presenters/helpers/censorship'
 
 module VCAP::CloudController
   class CNBLifecycleDataModel < Sequel::Model(:cnb_lifecycle_data)
     LIFECYCLE_TYPE = Lifecycles::CNB
+    set_field_as_encrypted :registry_credentials_json, salt: :credentials_salt, column: :encrypted_registry_credentials_json
```
[the remark this answers is not in the document]
Comment (pbusko) [reply app/models/runtime/cnb_lifecycle_data_model.rb]: done

File: spec/unit/lib/cloud_controller/diego/lifecycle_data_spec.rb (hunk 2 above)
Comment (pbusko) [reply spec/unit/lib/cloud_controller/diego/lifecycle_data_spec.rb:1]: good point, the file has been moved

**Pull request merged**